solr/solrj/src/java/org/apache/solr/common/cloud/ReplicaCount.java

```diff
@@ -61,6 +152,33 @@ public boolean contains(Replica.Type type) {
     return countByType.containsKey(type);
   }
 
+  /** Returns the replica types for which a number of replicas was explicitely defined. */
+  public Set<Replica.Type> keySet() {
+    return countByType.keySet();
+  }
+
+  /**
+   * Add properties for replica counts as integers to a properties map.
+   *
+   * @param propMap a properties map.
+   */
+  public void writeProps(Map<String, Object> propMap) {
+    for (Map.Entry<Replica.Type, Integer> entry : countByType.entrySet()) {
+      propMap.put(entry.getKey().numReplicasPropertyName, entry.getValue());
+    }
+  }
+
+  /**
+   * Add properties for replica counts as integers to Solr parameters.
+   *
+   * @param params a set of modifiable Solr parameters.
+   */
+  public void writeProps(ModifiableSolrParams params) {
+    for (Map.Entry<Replica.Type, Integer> entry : countByType.entrySet()) {
+      params.add(entry.getKey().numReplicasPropertyName, String.valueOf(entry.getValue()));
```

Comment:
More correct to use params.set, and you needn't add the valueOf part because `set` is overloaded with an int variant.

Reply:
Excellent, changed.